### PR TITLE
Add default header to clients indicating the client's request timeout value

### PR DESCRIPTION
**Pull request by chrisayoub**
cc @jhaber, @PaulFurtado

### CODE REVIEW

**Comment by jhaber** [comment]
cc @stevegutz @Xcelled @kmclarnon

File: HorizonNing/src/main/java/com/hubspot/horizon/ning/internal/DefaultHeadersRequestFilter.java [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+
+  private HttpConfig config;
+
+  public DefaultHeadersRequestFilter(HttpConfig config) {
+    this.config = config;
+  }
+
+  @Override
+  public <T> FilterContext<T> filter(FilterContext<T> context)  {
+    org.asynchttpclient.shaded.io.netty.handler.codec.http.HttpHeaders headers = context.getRequest().getHeaders();
```
Comment (stevegutz) [inline HorizonNing/src/main/java/com/hubspot/horizon/ning/internal/DefaultHeadersRequestFilter.java:19]: nit: why is this fully qualified? Can there be an import instead?

File: HorizonCore/src/main/java/com/hubspot/horizon/HorizonHeaders.java [hunk 2]
```diff
@@ -0,0 +1,7 @@
+package com.hubspot.horizon;
+
+public class HorizonHeaders {
+  public static final String REQUEST_TIMEOUT = "X-Horizon-Timeout-Millis";
+
+  private HorizonHeaders() {}
```
Comment (stevegutz) [inline HorizonCore/src/main/java/com/hubspot/horizon/HorizonHeaders.java:6]: nit: Can you throw an `AssertionError` in the body of this ctor?

**Comment by stevegutz** [comment]
It seems good overall. Is there more context on why we want this?

**Comment by PaulFurtado** [comment]
> It seems good overall. Is there more context on why we want this?

@stevegutz Our internal HTTP clients set `X-HubSpot-Timeout-Millis` which allows backends to give up on a request when the client has already timed out to avoid wasting effort processing the request. The java RainmakerClient uses Horizon, and we recently had a critsit where rainmaker was overwhelmed and every request was timing out due to hundreds of queued requests. So this change will allow us to drop any request that has sat in the request queue for longer than the timeout like our standard stack apps do

**Pull request merged**